access/gitlab/README.md

```diff
@@ -1,31 +1,89 @@
+GitLab projects. When someone requests new roles in Teleport, an issue will be opened,
 and the team members can assign approval or denied label to the issue to approve
 or deny the request.
 
-## Quick setup
+## Setup
 
-To get things up & running quickly:
+### Install the plugin
 
-1. On Gitlab, go "User Settings" -> "Access Tokens". Create a token with api
+Get the plugin distribution.
+
+```bash
+$ curl -L https://get.gravitational.com/teleport-access-gitlab-v7.0.2-linux-amd64-bin.tar.gz
+$ tar -xzf teleport-access-gitlab-v7.0.2-linux-amd64-bin.tar.gz
+$ cd teleport-access-gitlab
+$ ./install
+```
+
+### Set up GitLab project & API token
+
+1. On GitLab, go "User Settings" -> "Access Tokens". Create a token with api
    scope, remember the token.
 2. Create a project, get its numeric "Project ID" from "Project Overview" ->
    "Details" page.
 3. You might want to create the Board with lists: `Teleport: Pending`,
    `Teleport: Approved`, and `Teleport: Denied`. The plugin will work if you
    just change labels on issues, but with a Board you can just drag the issue
    into a status-column you want.
-4. Create an /etc/teleport-gitlab.yml
+
+### Teleport User and Role
+
+Using Web UI or `tctl` CLI utility, create the role `access-gitlab` and the user `access-gitlab` belonging to the role `access-gitlab`. You may use the following YAML declarations.
+
+#### Role
+
+```yaml
+kind: role
+metadata:
+  name: access-gitlab
+spec:
+  allow:
+    rules:
+      - resources: ['access_request']
+        verbs: ['list', 'read', 'update']
+version: v4
+```
+
+#### User
+
+```yaml
+kind: user
+metadata:
+  name: access-gitlab
+spec:
+  roles: ['access-gitlab']
+version: v2
+```
+
+### Generate the certificate
+
+For the plugin to connect to Auth Server, it needs an identity file containing TLS/SSH certificates. This can be obtained with tctl:
+
+```bash
+$ tctl auth sign --auth-server=AUTH-SERVER:PORT --format=file --user=access-gitlab --out=/var/lib/teleport/plugins/gitlab/auth_id --ttl=8760h
+```
+
+Here, `AUTH-SERVER:PORT` could be `localhost:3025`, `your-remote-auth.example.com:3025`, `your-remote-proxy.teleport.sh:443`. For remote connections, you might want to pass the `--identity=...` option to authenticate yourself to Auth Server.
+
+### Save configuration file
+
+By default, configuration file is expected to be at `/etc/teleport-gitlab.toml`.
 
 ```toml
 # /etc/teleport-gitlab.toml
 [teleport]
-auth_server = "example.com:3025"                         # Teleport Auth Server GRPC API address
-client_key = "/var/lib/teleport/plugins/gitlab/auth.key" # Teleport GRPC client secret key
-client_crt = "/var/lib/teleport/plugins/gitlab/auth.crt" # Teleport GRPC client certificate
-root_cas = "/var/lib/teleport/plugins/gitlab/auth.cas"   # Teleport cluster CA certs
+# Teleport Auth/Proxy Server address.
+#
+# Should be port 3080 or 443 for proxy.
+# For Teleport Cloud, should be in the form of "your-account.teleport.sh".
+auth_server = "example.com:3025"
```

Comment:
take a look at https://github.com/gravitational/teleport-plugins/tree/master/event-handler

the field `auth_server` is called `addr` there. Can you unify both field names? (e.g. support addr and deprecate auth_server by removing from the docs, but supporting it for transition)

Reply:
Done!

Slack, Mattermost and Pagerduty plugins are also updated. Jira plugin updated in #212